apps/src/code-studio/initApp/project.js

```diff
@@ -265,6 +267,19 @@ var projects = (module.exports = {
     return currentSources.makerAPIsEnabled;
   },
 
+  /**
+   * Calculates a md5 hash for everything within sources except the
+   * generatedProperties.
+   * @return {string} md5 hash string.
+   */
+  md5CurrentSources() {
+    const {
+      generatedProperties, // eslint-disable-line no-unused-vars
+      ...sourcesWithoutProperties
+    } = currentSources;
+    return MD5(JSON.stringify(sourcesWithoutProperties)).toString();
```

Comment:
Great candidate for future test coverage!

Reply:
(Verifying what properties _do_ and _don't_ cause a project to be viewed as "modified".)